clientcredentials/clientcredentials.go

```diff
@@ -94,6 +99,18 @@ func (c *tokenSource) Token() (*oauth2.Token, error) {
 	if len(c.conf.Scopes) > 0 {
 		v.Set("scope", strings.Join(c.conf.Scopes, " "))
 	}
+	// not client_secret nor auto_detect
+	if c.conf.AuthStyle > 2 {
+		var err error
+		if v, err = advancedauth.UrlValuesFromConfig(v, advancedauth.Config{
```

Comment:
this overrides `v` and removes all previous values (scope, grant_type). Is it intended? Maybe it should do some kind of merge with overrides?

Reply:
It actually doesn't.
`v` is passed to `UrlValuesFromConfig` and `v` is `url.Values` which is a type alias for `map[string][]string`

Since maps are passed v gets updated.

I'll refactor this to not return `v` as it's misleading.
Also maybe I'll change the name of the function to `ExtendUrlValues`

Reply:
Also, I had a version which returned vals without actually modifying the map but this required iterating over the returned map in `clientcredentials.go`.

I want to reduce the amount of custom code in the standard packages to minimum so I've decided to mutate in the function.